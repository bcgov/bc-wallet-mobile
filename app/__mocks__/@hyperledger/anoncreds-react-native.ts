const AnonCredsReactNative = {
  install: jest.fn(),
}

export default AnonCredsReactNative
