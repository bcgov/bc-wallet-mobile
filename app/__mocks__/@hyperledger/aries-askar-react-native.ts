const AriesAskarReactNative = {
  install: jest.fn(),
}

export default AriesAskarReactNative
