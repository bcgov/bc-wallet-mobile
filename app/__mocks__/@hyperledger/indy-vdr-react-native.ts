const IndyVDRReactNative = {
  install: jest.fn(),
}

export default IndyVDRReactNative
