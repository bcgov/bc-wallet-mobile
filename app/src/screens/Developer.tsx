import { useTheme, useStore, testIdWithKey, DispatchAction } from 'aries-bifold'
import React, { useState } from 'react'
import { useTranslation } from 'react-i18next'
import { Modal, SectionList, StyleSheet, Switch, Text, TouchableOpacity, View } from 'react-native'
import { SafeAreaView } from 'react-native-safe-area-context'
import Icon from 'react-native-vector-icons/MaterialIcons'

import { BCState } from '../store'

import IASEnvironment from './IASEnvironment'

interface Setting {
  title: string
  value?: string
  onPress?: () => void
  accessibilityLabel?: string
  testID?: string
}

interface SettingSection {
  header: {
    title: string
    icon: string
  }
  data: Setting[]
}

const Settings: React.FC = () => {
  const { t } = useTranslation()
  const [store, dispatch] = useStore<BCState>()
  const { SettingsTheme, TextTheme, ColorPallet } = useTheme()
  const [environmentModalVisible, setEnvironmentModalVisible] = useState<boolean>(false)
  const [devMode, setDevMode] = useState<boolean>(true)
  const [useVerifierCapability, setUseVerifierCapability] = useState<boolean>(!!store.preferences.useVerifierCapability)
  const [useConnectionInviterCapability, setConnectionInviterCapability] = useState(
    !!store.preferences.useConnectionInviterCapability
  )

  const styles = StyleSheet.create({
    container: {
      backgroundColor: ColorPallet.brand.primaryBackground,
      width: '100%',
    },
    section: {
      backgroundColor: SettingsTheme.groupBackground,
      paddingHorizontal: 25,
      paddingVertical: 24,
    },
    sectionHeader: {
      flexDirection: 'row',
      alignItems: 'center',
      paddingBottom: 0,
      marginBottom: -11,
    },
    sectionSeparator: {
      marginBottom: 10,
    },
    sectionRow: {
      flexDirection: 'row',
      alignItems: 'center',
      justifyContent: 'space-between',
    },
    itemSeparator: {
      borderBottomWidth: 1,
      borderBottomColor: ColorPallet.brand.primaryBackground,
      marginHorizontal: 25,
    },
    logo: {
      height: 64,
      width: '50%',
      marginVertical: 16,
    },
    footer: {
      marginVertical: 25,
      alignItems: 'center',
    },
  })

  const shouldDismissModal = () => {
    setEnvironmentModalVisible(false)
  }

  const settingsSections: SettingSection[] = [
    {
      header: {
        icon: 'apartment',
        title: 'IAS',
      },
      data: [
        {
          title: t('Developer.Environment'),
          value: store.developer.environment.name,
          accessibilityLabel: t('Developer.Environment'),
          testID: testIdWithKey('Environment'),
          onPress: () => {
            setEnvironmentModalVisible(true)
          },
        },
      ],
    },
  ]

  if (store.preferences.developerModeEnabled) {
    const section = settingsSections.find((item) => item.header.title === t('Settings.AppSettings'))
    if (section) {
      section.data = [
        ...section.data,
        {
          title: t('Settings.Developer'),
          accessibilityLabel: t('Settings.Developer'),
          testID: testIdWithKey('Developer'),
          onPress: () => {
            return
          },
        },
      ]
    }
  }

  const SectionHeader: React.FC<{ icon: string; title: string }> = ({ icon, title }) => (
    <View style={[styles.section, styles.sectionHeader]}>
      <Icon name={icon} size={24} style={{ marginRight: 10, color: TextTheme.normal.color }} />
      <Text style={[TextTheme.headingThree, { flexShrink: 1 }]}>{title}</Text>
    </View>
  )

  const SectionRow: React.FC<{
    title: string
    value?: string
    accessibilityLabel?: string
    testID?: string
    onPress?: () => void
  }> = ({ title, value, accessibilityLabel, testID, onPress, children }) => (
    <View style={[styles.section]}>
      <TouchableOpacity
        accessible={true}
        accessibilityLabel={accessibilityLabel}
        testID={testID}
        style={styles.sectionRow}
        onPress={onPress}
      >
        <Text style={[TextTheme.headingFour, { fontWeight: 'normal', maxWidth: '90%' }]}>{title}</Text>
        <Text style={[TextTheme.headingFour, { fontWeight: 'normal', maxWidth: '90%', color: ColorPallet.brand.link }]}>
          {value}
        </Text>
        {children}
      </TouchableOpacity>
    </View>
  )

  const toggleSwitch = () => {
    dispatch({
      type: DispatchAction.ENABLE_DEVELOPER_MODE,
      payload: [!devMode],
    })
    setDevMode(!devMode)
  }

  const toggleVerifierCapabilitySwitch = () => {
    dispatch({
      type: DispatchAction.USE_VERIFIER_CAPABILITY,
      payload: [!useVerifierCapability],
    })
    setUseVerifierCapability((previousState) => !previousState)
  }

  const toggleConnectionInviterCapabilitySwitch = () => {
    dispatch({
      type: DispatchAction.USE_CONNECTION_INVITER_CAPABILITY,
      payload: [!useConnectionInviterCapability],
    })
    setConnectionInviterCapability((previousState) => !previousState)
  }

  return (
    <SafeAreaView edges={['bottom', 'left', 'right']}>
      <Modal
        visible={environmentModalVisible}
        transparent={true}
        animationType={'slide'}
        onRequestClose={() => {
          return
        }}
      >
        <IASEnvironment shouldDismissModal={shouldDismissModal} />
      </Modal>
      <View style={styles.container}>
        <SectionRow title={t('Developer.DeveloperMode')}>
          <Switch
            accessibilityLabel={t('Developer.Toggle')}
            testID={testIdWithKey('ToggleDeveloper')}
            trackColor={{ false: ColorPallet.grayscale.lightGrey, true: ColorPallet.brand.primaryDisabled }}
            thumbColor={devMode ? ColorPallet.brand.primary : ColorPallet.grayscale.mediumGrey}
            ios_backgroundColor={ColorPallet.grayscale.lightGrey}
            onValueChange={toggleSwitch}
            value={devMode}
          />
        </SectionRow>
        <View style={[styles.sectionSeparator]}></View>
        <SectionList
          renderItem={({ item: { title, value, onPress } }) => (
            <SectionRow
              title={title}
              accessibilityLabel={title}
              testID={testIdWithKey(title.toLowerCase())}
              value={value}
              onPress={onPress}
            />
          )}
          renderSectionHeader={({
            section: {
              header: { title, icon },
            },
          }) => <SectionHeader icon={icon} title={title} />}
          ItemSeparatorComponent={() => (
            <View style={{ backgroundColor: SettingsTheme.groupBackground }}>
              <View style={[styles.itemSeparator]}></View>
            </View>
          )}
          SectionSeparatorComponent={() => <View style={[styles.sectionSeparator]}></View>}
          sections={settingsSections}
          stickySectionHeadersEnabled={false}
        ></SectionList>
        <SectionRow title={t('Verifier.UseVerifierCapability')}>
          <Switch
            accessibilityLabel={t('Verifier.Toggle')}
            testID={testIdWithKey('ToggleVerifierCapability')}
            trackColor={{ false: ColorPallet.grayscale.lightGrey, true: ColorPallet.brand.primaryDisabled }}
            thumbColor={useVerifierCapability ? ColorPallet.brand.primary : ColorPallet.grayscale.mediumGrey}
            ios_backgroundColor={ColorPallet.grayscale.lightGrey}
            onValueChange={toggleVerifierCapabilitySwitch}
            value={useVerifierCapability}
          />
        </SectionRow>
        <SectionRow title={t('Connection.UseConnectionInviterCapability')}>
          <Switch
            testID={testIdWithKey('ToggleConnectionInviterCapabilitySwitch')}
            trackColor={{ false: ColorPallet.grayscale.lightGrey, true: ColorPallet.brand.primaryDisabled }}
            thumbColor={useConnectionInviterCapability ? ColorPallet.brand.primary : ColorPallet.grayscale.mediumGrey}
            ios_backgroundColor={ColorPallet.grayscale.lightGrey}
            onValueChange={toggleConnectionInviterCapabilitySwitch}
            value={useConnectionInviterCapability}
          />
        </SectionRow>
      </View>
    </SafeAreaView>
  )
}

export default Settings
