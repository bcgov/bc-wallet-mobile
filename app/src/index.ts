import {
  translationResources,
  ConfigurationContext,
  types,
  Record,
  indyLedgers,
  defaultConfiguration,
} from 'aries-bifold'
import merge from 'lodash.merge'
import { ReducerAction } from 'react'

import bundles from './assets/branding/credential-branding'
import AddCredentialButton from './components/AddCredentialButton'
import AddCredentialSlider from './components/AddCredentialSlider'
import EmptyList from './components/EmptyList'
import HomeContentView from './components/HomeContentView'
import { useNotifications } from './hooks/notifications'
import en from './localization/en'
import { proofRequestTemplates } from './request-templates'
import Developer from './screens/Developer'
import { pages } from './screens/OnboardingPages'
import PersonCredential from './screens/PersonCredential'
import Splash from './screens/Splash'
import Terms from './screens/Terms'
import { BCDispatchAction } from './store'
import { defaultTheme as theme } from './theme'

const localization = merge({}, translationResources, {
  en: { translation: en },
})

const selectedLedgers = indyLedgers.filter((item) => !item.id.startsWith('Indicio'))
const configuration: ConfigurationContext = {
  ...defaultConfiguration,
  pages,
  splash: Splash,
  terms: Terms,
  homeContentView: HomeContentView,
  credentialListHeaderRight: AddCredentialButton,
  credentialListOptions: AddCredentialSlider,
  credentialEmptyList: EmptyList,
  developer: Developer,
  OCABundleResolver: new types.oca.OCABundleResolver(bundles as unknown as types.oca.Bundles),
  record: Record,
  indyLedgers: selectedLedgers,
  settings: [],
  customNotification: {
    component: PersonCredential,
    onCloseAction: (dispatch?: React.Dispatch<ReducerAction<any>>) => {
      if (dispatch) {
        dispatch({
          type: BCDispatchAction.PERSON_CREDENTIAL_OFFER_DISMISSED,
          payload: [{ personCredentialOfferDismissed: true }],
        })
      }
    },
    pageTitle: 'PersonCredential.PageTitle',
    title: 'PersonCredentialNotification.Title',
    description: 'PersonCredentialNotification.Description',
    buttonTitle: 'PersonCredentialNotification.ButtonTitle',
  },
  useCustomNotifications: useNotifications,
  proofRequestTemplates,
  enableTours: true,
}

export default { theme, localization, configuration }
