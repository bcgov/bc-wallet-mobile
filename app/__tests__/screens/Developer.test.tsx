import { AgentProvider, StoreProvider } from '@hyperledger/aries-bifold-core'
import { render } from '@testing-library/react-native'
import React from 'react'

import Developer from '../../src/screens/Developer'
import { AttestationProvider } from '../../src/services/attestation'
import { initialState, reducer } from '../../src/store'

const mockNavigation = jest.fn()
jest.mock('@react-navigation/native', () => ({
  ...jest.requireActual('@react-navigation/native'),
  useNavigation: () => {
    return mockNavigation
  },
}))

describe('Developer Screen', () => {
  beforeEach(() => {
    // Silence console.error because it will print a warning about Switch
    // "Warning: dispatchCommand was called with a ref ...".
    jest.spyOn(console, 'error').mockImplementation(() => {
      return null
    })
  })

  afterEach(() => {
    jest.clearAllMocks()
  })

  test('screen renders correctly', () => {
    const tree = render(
      <StoreProvider initialState={initialState} reducer={reducer}>
        <AttestationProvider>
          <Developer />
        </AttestationProvider>
      </StoreProvider>
    )

    expect(tree).toMatchSnapshot()
  })
})
