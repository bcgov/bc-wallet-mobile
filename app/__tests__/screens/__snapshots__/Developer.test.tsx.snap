// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Developer Screen screen renders correctly 1`] = `
<RNCSafeAreaView
  edges={
    [
      "bottom",
      "left",
      "right",
    ]
  }
>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={false}
    visible={false}
  />
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={false}
    visible={false}
  />
  <RCTScrollView
    style={
      {
        "backgroundColor": "#000000",
        "width": "100%",
      }
    }
  >
    <View>
      <View
        style={
          [
            {
              "backgroundColor": "#313132",
              "padding": 24,
            },
          ]
        }
      >
        <View
          style={
            {
              "flexDirection": "row",
            }
          }
        >
          <Text
            style={
              {
                "color": "#FFFFFF",
                "flex": 1,
                "flexWrap": "wrap",
                "fontSize": 21,
                "fontWeight": "normal",
              }
            }
          >
            Developer.DeveloperMode
          </Text>
          <View
            accessibilityLabel="Developer.Toggle"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
            testID="com.ariesbifold:id/ToggleDeveloper"
          >
            <RCTSwitch
              accessibilityRole="switch"
              onChange={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              onTintColor="rgba(53, 130, 63, 0.35)"
              style={
                [
                  {
                    "height": 31,
                    "width": 51,
                  },
                  {
                    "backgroundColor": "#D3D3D3",
                    "borderRadius": 16,
                  },
                ]
              }
              thumbTintColor="#42803E"
              tintColor="#D3D3D3"
              value={true}
            />
          </View>
        </View>
      </View>
      <View
        style={
          [
            {
              "marginBottom": 10,
            },
          ]
        }
      />
      <View
        style={
          [
            {
              "backgroundColor": "#313132",
              "padding": 24,
            },
            {
              "alignItems": "center",
              "flexDirection": "row",
              "paddingBottom": 0,
            },
          ]
        }
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            [
              {
                "color": undefined,
                "fontSize": 24,
              },
              {
                "color": "#FFFFFF",
                "marginRight": 10,
              },
              {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          î©€
        </Text>
        <Text
          style={
            [
              {
                "color": "#FFFFFF",
                "fontSize": 26,
                "fontWeight": "bold",
              },
              {
                "flexShrink": 1,
              },
            ]
          }
        >
          IAS
        </Text>
      </View>
      <View
        style={
          [
            {
              "backgroundColor": "#313132",
              "padding": 24,
            },
          ]
        }
      >
        <View
          style={
            {
              "flexDirection": "row",
            }
          }
        >
          <Text
            style={
              {
                "color": "#FFFFFF",
                "flex": 1,
                "flexWrap": "wrap",
                "fontSize": 21,
                "fontWeight": "normal",
              }
            }
          >
            Developer.Environment
          </Text>
          <View
            accessibilityLabel="Developer.Environment"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
            testID="com.ariesbifold:id/developer.environment"
          >
            <Text
              style={
                [
                  {
                    "color": "#FFFFFF",
                    "fontSize": 21,
                    "fontWeight": "bold",
                  },
                  {
                    "color": "#FFFFFF",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              Production
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          [
            {
              "marginBottom": 10,
            },
          ]
        }
      />
      <View
        style={
          [
            {
              "backgroundColor": "#313132",
              "padding": 24,
            },
          ]
        }
      >
        <View
          style={
            {
              "flexDirection": "row",
            }
          }
        >
          <Text
            style={
              {
                "color": "#FFFFFF",
                "flex": 1,
                "flexWrap": "wrap",
                "fontSize": 21,
                "fontWeight": "normal",
              }
            }
          >
            Verifier.UseVerifierCapability
          </Text>
          <View
            accessibilityLabel="Verifier.Toggle"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
            testID="com.ariesbifold:id/ToggleVerifierCapability"
          >
            <RCTSwitch
              accessibilityRole="switch"
              onChange={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              onTintColor="rgba(53, 130, 63, 0.35)"
              style={
                [
                  {
                    "height": 31,
                    "width": 51,
                  },
                  {
                    "backgroundColor": "#D3D3D3",
                    "borderRadius": 16,
                  },
                ]
              }
              thumbTintColor="#606060"
              tintColor="#D3D3D3"
              value={false}
            />
          </View>
        </View>
      </View>
      <View
        style={
          {
            "backgroundColor": "#313132",
          }
        }
      >
        <View
          style={
            [
              {
                "borderBottomColor": "#000000",
                "borderBottomWidth": 1,
                "marginHorizontal": 24,
              },
            ]
          }
        />
      </View>
      <View
        style={
          [
            {
              "backgroundColor": "#313132",
              "padding": 24,
            },
          ]
        }
      >
        <View
          style={
            {
              "flexDirection": "row",
            }
          }
        >
          <Text
            style={
              {
                "color": "#FFFFFF",
                "flex": 1,
                "flexWrap": "wrap",
                "fontSize": 21,
                "fontWeight": "normal",
              }
            }
          >
            Verifier.AcceptDevCredentials
          </Text>
          <View
            accessibilityLabel="Verifier.Toggle"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
            testID="com.ariesbifold:id/ToggleAcceptDevCredentials"
          >
            <RCTSwitch
              accessibilityRole="switch"
              onChange={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              onTintColor="rgba(53, 130, 63, 0.35)"
              style={
                [
                  {
                    "height": 31,
                    "width": 51,
                  },
                  {
                    "backgroundColor": "#D3D3D3",
                    "borderRadius": 16,
                  },
                ]
              }
              thumbTintColor="#606060"
              tintColor="#D3D3D3"
              value={false}
            />
          </View>
        </View>
      </View>
      <View
        style={
          {
            "backgroundColor": "#313132",
          }
        }
      >
        <View
          style={
            [
              {
                "borderBottomColor": "#000000",
                "borderBottomWidth": 1,
                "marginHorizontal": 24,
              },
            ]
          }
        />
      </View>
      <View
        style={
          [
            {
              "backgroundColor": "#313132",
              "padding": 24,
            },
          ]
        }
      >
        <View
          style={
            {
              "flexDirection": "row",
            }
          }
        >
          <Text
            style={
              {
                "color": "#FFFFFF",
                "flex": 1,
                "flexWrap": "wrap",
                "fontSize": 21,
                "fontWeight": "normal",
              }
            }
          >
            Connection.UseConnectionInviterCapability
          </Text>
          <View
            accessibilityLabel="Connection.Toggle"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
            testID="com.ariesbifold:id/ToggleConnectionInviterCapabilitySwitch"
          >
            <RCTSwitch
              accessibilityRole="switch"
              onChange={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              onTintColor="rgba(53, 130, 63, 0.35)"
              style={
                [
                  {
                    "height": 31,
                    "width": 51,
                  },
                  {
                    "backgroundColor": "#D3D3D3",
                    "borderRadius": 16,
                  },
                ]
              }
              thumbTintColor="#606060"
              tintColor="#D3D3D3"
              value={false}
            />
          </View>
        </View>
      </View>
      <View
        style={
          {
            "backgroundColor": "#313132",
          }
        }
      >
        <View
          style={
            [
              {
                "borderBottomColor": "#000000",
                "borderBottomWidth": 1,
                "marginHorizontal": 24,
              },
            ]
          }
        />
      </View>
      <View
        style={
          [
            {
              "backgroundColor": "#313132",
              "padding": 24,
            },
          ]
        }
      >
        <View
          style={
            {
              "flexDirection": "row",
            }
          }
        >
          <Text
            style={
              {
                "color": "#FFFFFF",
                "flex": 1,
                "flexWrap": "wrap",
                "fontSize": 21,
                "fontWeight": "normal",
              }
            }
          >
            Verifier.UseDevVerifierTemplates
          </Text>
          <View
            accessibilityLabel="Verifier.ToggleDevTemplates"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
            testID="com.ariesbifold:id/ToggleDevVerifierTemplatesSwitch"
          >
            <RCTSwitch
              accessibilityRole="switch"
              onChange={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              onTintColor="rgba(53, 130, 63, 0.35)"
              style={
                [
                  {
                    "height": 31,
                    "width": 51,
                  },
                  {
                    "backgroundColor": "#D3D3D3",
                    "borderRadius": 16,
                  },
                ]
              }
              thumbTintColor="#606060"
              tintColor="#D3D3D3"
              value={false}
            />
          </View>
        </View>
      </View>
      <View
        style={
          {
            "backgroundColor": "#313132",
          }
        }
      >
        <View
          style={
            [
              {
                "borderBottomColor": "#000000",
                "borderBottomWidth": 1,
                "marginHorizontal": 24,
              },
            ]
          }
        />
      </View>
      <View
        style={
          [
            {
              "backgroundColor": "#313132",
              "padding": 24,
            },
          ]
        }
      >
        <View
          style={
            {
              "flexDirection": "row",
            }
          }
        >
          <Text
            style={
              {
                "color": "#FFFFFF",
                "flex": 1,
                "flexWrap": "wrap",
                "fontSize": 21,
                "fontWeight": "normal",
              }
            }
          >
            NameWallet.EnableWalletNaming
          </Text>
          <View
            accessibilityLabel="NameWallet.ToggleWalletNaming"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
            testID="com.ariesbifold:id/ToggleWalletNamingSwitch"
          >
            <RCTSwitch
              accessibilityRole="switch"
              onChange={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              onTintColor="rgba(53, 130, 63, 0.35)"
              style={
                [
                  {
                    "height": 31,
                    "width": 51,
                  },
                  {
                    "backgroundColor": "#D3D3D3",
                    "borderRadius": 16,
                  },
                ]
              }
              thumbTintColor="#606060"
              tintColor="#D3D3D3"
              value={false}
            />
          </View>
        </View>
      </View>
      <View
        style={
          {
            "backgroundColor": "#313132",
          }
        }
      >
        <View
          style={
            [
              {
                "borderBottomColor": "#000000",
                "borderBottomWidth": 1,
                "marginHorizontal": 24,
              },
            ]
          }
        />
      </View>
      <View
        style={
          [
            {
              "backgroundColor": "#313132",
              "padding": 24,
            },
          ]
        }
      >
        <View
          style={
            {
              "flexDirection": "row",
            }
          }
        >
          <Text
            style={
              {
                "color": "#FFFFFF",
                "flex": 1,
                "flexWrap": "wrap",
                "fontSize": 21,
                "fontWeight": "normal",
              }
            }
          >
            Settings.PreventAutoLock
          </Text>
          <View
            accessibilityLabel="Settings.TogglePreventAutoLock"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
            testID="com.ariesbifold:id/TogglePreventAutoLockSwitch"
          >
            <RCTSwitch
              accessibilityRole="switch"
              onChange={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              onTintColor="rgba(53, 130, 63, 0.35)"
              style={
                [
                  {
                    "height": 31,
                    "width": 51,
                  },
                  {
                    "backgroundColor": "#D3D3D3",
                    "borderRadius": 16,
                  },
                ]
              }
              thumbTintColor="#606060"
              tintColor="#D3D3D3"
              value={false}
            />
          </View>
        </View>
      </View>
      <View
        style={
          {
            "backgroundColor": "#313132",
          }
        }
      >
        <View
          style={
            [
              {
                "borderBottomColor": "#000000",
                "borderBottomWidth": 1,
                "marginHorizontal": 24,
              },
            ]
          }
        />
      </View>
      <View
        style={
          [
            {
              "backgroundColor": "#313132",
              "padding": 24,
            },
          ]
        }
      >
        <View
          style={
            {
              "flexDirection": "row",
            }
          }
        >
          <Text
            style={
              {
                "color": "#FFFFFF",
                "flex": 1,
                "flexWrap": "wrap",
                "fontSize": 21,
                "fontWeight": "normal",
              }
            }
          >
            Remote Logging
          </Text>
          <View
            accessibilityLabel="Remote Logging"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
            testID="com.ariesbifold:id/ToggleRemoteLoggingSwitch"
          >
            <RCTSwitch
              accessibilityRole="switch"
              disabled={true}
              onChange={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              onTintColor="rgba(53, 130, 63, 0.35)"
              style={
                [
                  {
                    "height": 31,
                    "width": 51,
                  },
                  {
                    "backgroundColor": "#D3D3D3",
                    "borderRadius": 16,
                  },
                ]
              }
              thumbTintColor="#606060"
              tintColor="#D3D3D3"
              value={false}
            />
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RNCSafeAreaView>
`;
